kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Queue.scala

```diff
@@ -19,83 +19,101 @@ final private class Queue[T](implicit ord: Ordering[T]) extends AtomicBoolean {
         items
     }
 
-    def add(value: T): Unit =
-        modify {
+    def add(value: T): Unit = {
+        lock()
+        try {
             items += 1
-            queue.enqueue(value)
```

Comment:
This code was using `addOne` before the cross compilation to Scala 2.12, which doesn't have this method. I had changed it to `enqueue` but the underlying implementation allocates an `ArrayRef` and uses scala collection methods that have more overhead. The `+=` method is equivalent to `addOne`.